autoload/cmake/fileapi.vim

```diff
@@ -0,0 +1,188 @@
+" ==============================================================================
+" Location:    autoload/cmake/cmake-api.vim
+" Description: Functions for interfacing with the cmake-fila-api(7)
+" ==============================================================================
+
+let s:fileapi = {}
+let s:fileapi.cmake_version_supported = v:false
+
+let s:client_name = 'vim-cmake'
+let s:api_path = ['.cmake', 'api', 'v1']
+let s:query_path = s:api_path + ['query', 'client-vim-cmake', 'query.json']
+let s:reply_path = s:api_path + ['reply']
+
+let s:query_version = 1
+let s:query = {
+            \     'requests': [
+            \         { 'kind': 'codemodel' , 'version': 2 },
+            \     ],
+            \     'client': {
+            \         'query': { 'version': s:query_version },
+            \     },
+            \ }
+
+let s:logger = cmake#logger#Get()
+let s:system = cmake#system#Get()
+
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+" Private functions
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+" Find the current index file
+"
+function! s:FindIndexFile(build_dir) abort
+    let l:glob = s:system.Path([a:build_dir] + s:reply_path + ['index'], v:true) . '-*.json'
```

Comment:
I hadn't thought of that. Although it probably doesn't matter too much in this file since this information is either short lived or non crucial. Will change anyway.

Reply:
Actually the result of `s:FindIndexFile()` is not short lived, it gets saved into `s:fileapi.last_index_name`. If, after running this once, we `:cd` into a subfolder of the root folder, and obtain the new relative path to the index file, `s:ParseIndex()` will think the new file name differs from the previous one.

Reply:
While the result is not short lived it is only used in a caching manner. Changing directory would essentially just invalidate the cache. Nothing would break because of this (most likely, CMake names its index file with a timestamp down to the millisecond). In any case, it is more correct to save the full path.